diff -Naur a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel	2022-08-09 13:12:44.000000000 -0400
+++ b/BUILD.bazel	2022-09-06 00:26:04.407595668 -0400
@@ -877,6 +877,40 @@
     }),
 )
 
+cc_library(
+    name = "pyext",
+    visibility = ["//visibility:public"],
+    srcs = glob([
+        "python/google/protobuf/pyext/*.cc",
+        "python/google/protobuf/pyext/*.h",
+    ]),
+    copts = COPTS + [
+        "-DGOOGLE_PROTOBUF_HAS_ONEOF=1",
+    ] + select({
+        "//conditions:default": [],
+        ":allow_oversize_protos": ["-DPROTOBUF_PYTHON_ALLOW_OVERSIZE_PROTOS=1"],
+    }),
+    includes = [
+        "python/",
+        "src/",
+    ],
+    linkstatic = 1,
+    tags = [
+        # Exclude this target from wildcard expansion (//...) because it may
+        # not even be buildable. It will be built if it is needed according
+        # to :use_fast_cpp_protos.
+        # https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes
+        "manual",
+    ],
+    deps = [
+        ":protobuf",
+        ":proto_api",
+    ] + select({
+        "//conditions:default": [],
+        ":use_fast_cpp_protos": ["//external:python_headers"],
+    }),
+)
+
 config_setting(
     name = "use_fast_cpp_protos",
     values = {
