diff -r c8b3a84e2006 beancount/parser/lexer.c
--- a/beancount/parser/lexer.c	Tue May 01 23:18:54 2018 -0400
+++ b/beancount/parser/lexer.c	Tue May 01 23:20:14 2018 -0400
@@ -1068,7 +1068,7 @@
 	if ( YY_CURRENT_BUFFER_LVALUE->yy_is_interactive ) \
 		{ \
 		int c = '*'; \
-		size_t n; \
+		ssize_t n; \
 		for ( n = 0; n < max_size && \
 			     (c = getc( yyin )) != EOF && c != '\n'; ++n ) \
 			buf[n] = (char) c; \
